File: binary_search/34_find_first_and_last_position_of_element_in_sorted_array.go
func searchRange(nums []int, target int) []int {

    l, r, mid := 0, len(nums)-1, 0
    found := false

    for l <= r {
        mid = (l+r) / 2

        if nums[mid] == target {
            found = true
            break
        } else if nums[mid] < target {
            l = mid + 1
        } else {
            r = mid - 1
        }
    }

    if !found {
        return []int{-1,-1}
    }

    left, right := mid, mid

    for left > 0 && nums[left-1] == target {
        left --
    }

    for right < len(nums)-1 && nums[right+1] == target {
        right ++
    }

    return []int{left, right}
}

File: binary_search/34_find_first_and_last_position_of_element_in_sorted_array.js
/**
 * @param {number[]} nums
 * @param {number} target
 * @return {number[]}
 */
var searchRange = function(nums, target) {

    let l = 0
    let r = nums.length - 1
    let found = false
    let mid

    while(l <= r){

        mid = Math.floor((l+r) / 2)

        if(nums[mid] === target){
            found = true
            break
        }
        else if(nums[mid] > target) r = mid -1
        else l = mid + 1

    }

    if(!found) return [-1, -1]

    let left = mid
    let right = mid

    while((0 <= left-1) && (nums[left-1] === target)) left -= 1
    while((nums.length-1 >= right+1) && (nums[right+1] === target)) right += 1

    return [left, right]

};